aws/resource_aws_s3_bucket_test.go

```diff
@@ -899,7 +902,7 @@ func TestAWSS3BucketName(t *testing.T) {
 }
 
 func testAccCheckAWSS3BucketDestroy(s *terraform.State) error {
-	return testAccCheckInstanceDestroyWithProvider(s, testAccProvider)
+	return testAccCheckAWSS3BucketDestroyWithProvider(s, testAccProvider)
```

Comment:
😱

Reply:
😢  Unfortunately, I cannot think of a good solution off the top of my head to this problem unless we split everything into separate Go packages in the provider. Even then, you could still typo something like `S3Bucket` vs `S3BucketPolicy`, etc.